### PR TITLE
[Feature] Support nuimages visualization

### PR DESCRIPTION
**Pull request by xiliu8006**
Support nuimages visualization

### CODE REVIEW

File: mmdet3d/apis/test.py [hunk 1]
```diff
@@ -29,8 +35,38 @@ def single_gpu_test(model, data_loader, show=False, out_dir=None):
             result = model(return_loss=False, rescale=True, **data)
 
         if show:
-            model.module.show_results(data, result, out_dir)
+            if hasattr(model.module, 'show_results'):
+                model.module.show_results(data, result, out_dir)
+            else:
+                batch_size = len(result)
```
Comment (ZwwWayne) [inline mmdet3d/apis/test.py:41]: Add comment here indicating where is this logic adopted from and what is it for.

File: mmdet3d/apis/test.py [hunk 2]
```diff
@@ -29,8 +35,38 @@ def single_gpu_test(model, data_loader, show=False, out_dir=None):
             result = model(return_loss=False, rescale=True, **data)
 
         if show:
-            model.module.show_results(data, result, out_dir)
+            if hasattr(model.module, 'show_results'):
+                model.module.show_results(data, result, out_dir)
```
Comment (Tai-Wang) [inline mmdet3d/apis/test.py:39]: `show_results` and `show_result` are so confusing here. Please at least add a comment to explain the former is for visualization in mmdet3d (maybe multi-modality visualization in the future).

**Review by Tai-Wang — CHANGES_REQUESTED** [review]
Just need to add some supplementary comments.

File: mmdet3d/apis/test.py [hunk 3]
```diff
@@ -29,8 +35,42 @@ def single_gpu_test(model, data_loader, show=False, out_dir=None):
             result = model(return_loss=False, rescale=True, **data)
 
         if show:
-            model.module.show_results(data, result, out_dir)
+            # Visualize the results of MMdetection3D model
+            # 'show_results' is MMdetection3D visualization API
+            if hasattr(model.module, 'show_results'):
```
Comment (ZwwWayne) [inline mmdet3d/apis/test.py:40]: Should use `isinstance` to determine whether the detector is a 2D detector or a 3D detector.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**